packages/wonder-blocks-dropdown/components/single-select.js

```diff
@@ -175,36 +194,90 @@ export default class SingleSelect extends React.Component<Props, State> {
         }
     };
 
-    getMenuItems(): Array<DropdownItem> {
-        const {children, selectedValue} = this.props;
+    mapOptionItemsToDropdownItems = (
+        children: Array<React.Element<OptionItem>>,
+    ): Array<DropdownItem> => {
         // Figure out which index should receive focus when this select opens
         // Needs to exclude counting items that are disabled
         let indexCounter = 0;
         this.selectedIndex = 0;
 
-        return React.Children.toArray(children)
-            .filter(Boolean)
-            .map((option) => {
-                const {disabled, value} = option.props;
-                const selected = selectedValue === value;
-                if (selected) {
-                    this.selectedIndex = indexCounter;
-                }
-                if (!disabled) {
-                    indexCounter += 1;
-                }
-                return {
-                    component: option,
-                    focusable: !disabled,
-                    populatedProps: {
-                        onToggle: this.handleToggle,
-                        selected: selected,
-                        variant: "check",
-                    },
-                };
-            });
+        return children.map((option) => {
+            const {selectedValue} = this.props;
+            const {disabled, value} = option.props;
+            const selected = selectedValue === value;
+
+            if (selected) {
+                this.selectedIndex = indexCounter;
+            }
+            if (!disabled) {
+                indexCounter += 1;
+            }
+            return {
+                component: option,
+                focusable: !disabled,
+                populatedProps: {
+                    onToggle: this.handleToggle,
+                    selected: selected,
+                    variant: "check",
+                },
+            };
+        });
+    };
+
+    getMenuItems(
+        children: Array<React.Element<OptionItem>>,
+    ): Array<DropdownItem> {
+        const {isFilterable} = this.props;
+
+        // If it's not filterable, no need to do any extra besides mapping the
+        // option items to dropdown items.
+        if (!isFilterable) {
+            this.mapOptionItemsToDropdownItems(children);
+        }
+
+        const {searchText} = this.state;
+
+        const lowercasedSearchText = searchText.toLowerCase();
+
+        // Filter the children with the searchText if any.
+        const filteredChildren = children.filter(
+            ({props}) =>
+                !searchText ||
+                props.label.toLowerCase().indexOf(lowercasedSearchText) > -1,
+        );
+
+        return this.mapOptionItemsToDropdownItems(filteredChildren);
```

Comment:
style nit: I think this function could be simplified a bit like so:
```
return this.mapOptionItemsToDropdownItems(
  isFilterable ? this.filterChildren(children) : children
);
```
with the filtering logic living in a separate method.